.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: CI build for the morandi-rb gem
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+
+jobs:
```

Comment:
steps are run top to bottom, but jobs are run separately. the matrix build (i.e., the run of the specs against the three target ruby versions) also defaults to [fail-fast](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) (i.e., if it fails on one target version then the tests for the other two will exit early). however this can be configured if necessary.